src/main/java/com/cloudbees/jenkins/support/api/BaseFileContent.java

```diff
@@ -124,8 +124,9 @@ protected void writeTo(OutputStream os) throws IOException {
     }
 
     protected void writeTo(OutputStream os, @NonNull ContentFilter filter) throws IOException {
-        if (isBinary) {
+        if (isBinary || filter == ContentFilter.NONE) {
```

Comment:
If we are using `ContentFilter.NONE`, just write everything directly as binary.

It would be nice to preserve newlines in all cases, but the use of `BufferedReader.readLine` below makes that difficult to do, and anyways, once anonymization is involved, any debugging that cares about byte-level details is not going work in the first place so I don't think it's that important.